distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso

```diff
@@ -0,0 +1,582 @@
+        handle_java_exception <|
+            case mc of
+                _ : Math_Context -> Decimal.Value <| handle_precision_loss i <| Decimal_Utils.fromInteger i mc.math_context
+                _ : Nothing -> Decimal.Value (Decimal_Utils.fromInteger i)
+
+    ## GROUP conversions
+       Construct a `Decimal` from a `Float`.
+
+       Arguments:
+       - f: The `Float` to construct a `Decimal` from.
+       - mc: The `Math_Context` to use to specify precision and `Rounding_Mode`.
+         If a `Math_Context` is used, there is a possibility of a loss of
+         precision.
+
+       ? Precision Loss
+
+         A `Decimal` can represent any `Float` precisely. However, some `Float`
+         literals, such as `0.1`, cannot be represented exactly to infinite
+         precision by a `Float`. For this reason, constructing a `Decimal` from
+         a `Float` always attaches a `Loss_Of_Numeric_Precision` warning to the
+         result.
+
+       ! Error Conditions
+
+         - A `Loss_Of_Numeric_Precision` warning is always attached when
+           converting to `Decimal` from `Float`.
+         - If `f` is NaN or +/-Inf, an Illegal_Argument error is thrown.
+
+        ^ Example
+          Create a `Decimal` from a float.
+
+              d = Decimal.from_integer 12.345
+    from_float : Float -> Math_Context -> Decimal ! Arithmetic_Error ! Loss_Of_Numeric_Precision ! Illegal_Argument
+    from_float (f : Float) (mc : Math_Context | Nothing)=Nothing -> Decimal ! Loss_Of_Numeric_Precision ! Illegal_Argument =
+        is_exceptional = f.is_nan || f.is_infinite
+        if is_exceptional then Error.throw (Illegal_Argument.Error "Cannot convert "+f.to_text+" to a Decimal") else
+            # We do not use `handle_precision_loss` here because we always attach a `Loss_Of_Numeric_Precision` warning in the Float case.
+            handle_java_exception <| attach_loss_of_numeric_precision f <|
+                case mc of
+                    _ : Math_Context -> Decimal.Value <| Decimal_Utils.fromFloat f mc.math_context
+                    _ : Nothing -> Decimal.Value (Decimal_Utils.fromFloat f)
+
+    ## ALIAS greater than
+       GROUP Operators
+       ICON operators
+       Checks if this is greater than that.
+
+       Arguments:
+       - that: The number to compare this against.
+
+       > Example
+         Checking if 10 is greater than 7.
+
+             Decimal.new 10 > 7
+             # => True
+    > : Decimal -> Boolean
+    > self (that:Decimal) -> Boolean = Decimal_Comparator.compare self that == Ordering.Greater
+
+    ## ALIAS greater than or equal
+       GROUP Operators
+       ICON operators
+       Checks if this is greater than or equal to that.
+
+       Arguments:
+       - that: The number to compare this against.
+
+       > Example
+         Checking if 10 is greater than or equal to 7.
+
+             Decimal.new 10 >= 7
+             # => True
+    >= : Decimal -> Boolean
+    >= self (that:Decimal) -> Boolean =
```

Comment:
Do we need these or will the base ones on Any do the same thing?

Reply:
No, without these, we get `Incomparable_Values`. (@JaroslavTulach, this is expected behavior, right?)

Reply:
> No, without these, we get `Incomparable_Values`. (@JaroslavTulach, this is expected behavior, right?)

Yes, I believe that is expected behavior, because:

> Do we need these or will the base ones on Any do the same thing?

There shall be **no `<=`, `<` or `>=`, `>` on `Any` to begin with**! Unlike `==` (which is supported for all Enso values), _inequalities aren't defined_ for all Enso values. Having them on `Any` is a mistake which has practical implications - the component browser gets polluted by these operations even for types where they make no sense (unless a special care is taken to remove them).

Definition of `<=`, `<`, `>=`, `>` shall be removed from `Any` and each type willing to provide _inequality ordering_ shall define them themselves, properly specifying the `that` type it can be compared to.